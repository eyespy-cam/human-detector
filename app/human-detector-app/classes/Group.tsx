import * as React from 'react';
import { View } from 'react-native';
import { StatusBar } from 'expo-status-bar';
import Camera from './Camera';


export default class Group{
    groupName:string
    groupID:string
    cameras:Camera[]

    constructor(groupName:string, groupID:string){
        this.groupName = groupName
        this.groupID = groupID
        this.cameras = new Array()
    }
}

export function renameGroup(newName:string, group:Group): Group {
    return new Group('name', 'ID')       
}

export function addToGroup(newCam:Camera, group:Group): Group {
    return new Group('name', 'ID') 
}