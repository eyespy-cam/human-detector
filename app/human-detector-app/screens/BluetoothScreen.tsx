import * as React from 'react';
import { View, Text, ScrollView, Button, Alert } from 'react-native';
import useBLE from '../src/ble/bletest';

export default function BluetoothScreen(): React.ReactElement {
    
    // Make list of all bluetooth devices

    const {
        requestPermissions,
        connectToDevice,
        scanForDevices,
        currentDevice,
        heartRate,
        allDevices,
      } = useBLE();
    
      requestPermissions((isGranted: boolean) => {
        if (isGranted) {
          scanForDevices();
        }
      });
    
      console.log(allDevices);

    return (
    <View>
        <ScrollView>
            {allDevices.map((item) => (
                <View key={item.id}>
                    <Button 
                    title={item.id}
                    onPress={() => Alert.alert('Connection Successful')}/>
                </View>
            ))}
        </ScrollView>
    </View>
    )
}