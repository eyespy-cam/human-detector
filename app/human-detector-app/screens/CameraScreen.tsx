import * as React from 'react';
import { useState } from 'react';
import { ScrollView, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import CameraSettingsButton from '../components/CameraSettingsButton';
import Camera from '../classes/Camera';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    // paddingTop: 40,
    // paddingHorizontal: 20
  },
  textInput: {
    borderWidth: 1,
    borderColor: '#777',
    padding: 8,
  },
  pads: {
    padding: 10,
  },
  boldHeader: {
    fontWeight: 'bold',
    fontSize: 20,
  },
  menuItem: {
    marginTop: 24,
    marginLeft: 20,
    marginRight: 20,
    padding: 30,
    backgroundColor: '#E0FFFF',
    fontSize: 24,
    borderWidth: 2,
    borderColor: '#D3D3D3',
  },
  addButtonItem: {
    borderColor: '#D3D3D3',
    backgroundColor: '#DCDCDC',
  },
  menuButtonText: {
    fontSize: 24,
    marginTop: 10,
    marginBottom: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  addButtonText: {
    alignItems: 'center',
    fontSize: 50,
    marginTop: 0,
    marginBottom: 0,
  },
});

export default function CameraScreen({ navigation }): React.ReactElement {
  const cameraOne: Camera = new Camera('123', "AAAAA's Camera", '99');
  const cameraTwo: Camera = new Camera('124', "BBBBB's Camera", '725');
  const cameraThree: Camera = new Camera('125', "CCCCC's Camera", '400');

  const [listOfCameras, setListOfCameras] = useState([cameraOne, cameraTwo, cameraThree]);

  return (
    <View style={styles.container}>
      <ScrollView>
        {listOfCameras.map((item) => (
          <View key={item.cameraId}>
            <TouchableOpacity
              style={styles.menuItem}
              onPress={() => {
                setListOfCameras([item]);
              }}
            >
              <Text style={styles.menuButtonText}> {item.cameraName} </Text>
              <CameraSettingsButton cameraId={item.cameraId} />
            </TouchableOpacity>
          </View>
        ))}

        <View key="add-button">
          <TouchableOpacity
            style={[styles.menuItem, styles.addButtonItem]}
            onPress={() => {
              // Start camera registration process
              navigation.navigate('Bluetooth');
            }}
          >
            <Text style={styles.addButtonText}> + </Text>
          </TouchableOpacity>
        </View>
      </ScrollView>
    </View>
  );
}

export function CameraOnPress() {}

export function CameraDisplayButton() {}

export function getCameraThumbnail() {}

export function getCameraStream() {}

export function isCameraOnline(): boolean {
  return true;
}
