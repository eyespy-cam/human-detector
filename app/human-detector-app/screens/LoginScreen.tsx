import * as React from 'react';
import { View, StyleSheet, Button } from 'react-native';
import { makeRedirectUri, useAuthRequest, ResponseType } from 'expo-auth-session';
import * as AuthSession from 'expo-auth-session';
import Constants from 'expo-constants';
import { generateRandom } from 'expo-auth-session/build/PKCE';

export default function LoginScreen(): React.ReactElement {
  // eslint-disable-next-line no-unsafe-optional-chaining
  const apiUrl: string = Constants.manifest?.extra?.keycloakUrl;

  const discovery = AuthSession.useAutoDiscovery(`${apiUrl}/realms/myrealm`);

  const [request, response, promptAsync] = useAuthRequest(
    {
      clientId: 'myclient',
      scopes: ['openid'],
      codeChallenge: generateRandom(30),
      redirectUri: makeRedirectUri({}),
    },
    discovery
  );

  React.useEffect(() => {
    if (response?.type === 'success') {
      const { code } = response.params;
      console.log(code);
    }
  }, [response]);

  // registerNNPushToken(4044, 'bbBo3vbjuqXCetI9mvFYic');
  // console.log(DeviceInfo.getModel()); // it returns 'Simulator'

  return (
    <View>
      <Button
        onPress={() => {
          promptAsync();
        }}
        title="Login with KeyCloak"
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    // paddingTop: 40,
    // paddingHorizontal: 20
  },
});
