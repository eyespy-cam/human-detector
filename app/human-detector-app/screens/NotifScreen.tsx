import * as React from 'react';
import { StatusBar } from 'expo-status-bar';
import { View } from 'react-native';

export default function NotifScreen(){

}

export function ListNotifs(){

}