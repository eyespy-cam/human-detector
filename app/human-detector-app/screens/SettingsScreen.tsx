export default function SettingsScreen() {}

export function ListSettings() {}
