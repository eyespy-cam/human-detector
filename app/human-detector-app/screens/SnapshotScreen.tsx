import { NativeStackScreenProps } from '@react-navigation/native-stack';
import * as React from 'react';
import { View, Image, ImageURISource, Dimensions } from 'react-native';
import ImageZoom from 'react-native-image-pan-zoom';
import { LoadingIcon, LoadingState } from '../components/LoadingIcon';
import { BackendContext } from '../contexts/backendContext';
import { RootStackParamList } from '../src/navigation/stackParamList';

type Props = NativeStackScreenProps<RootStackParamList, 'Snapshot'>;

export default function SnapshotScreen({ route }: Props): React.ReactElement<Props> {
  const { snapshotId } = route.params;
  const backendService = React.useContext(BackendContext);
  const [snapshotUrl, setSnapshotUrl] = React.useState<ImageURISource | null>(null);

  React.useEffect(() => {
    backendService!
      .getSnapshotSource(snapshotId)
      .then(setSnapshotUrl)
      .catch((error) => console.error(`Failed to fetch snapshot URL`, error));
  }, []);

  if (!snapshotUrl) {
    // eslint-disable-next-line react/jsx-no-useless-fragment
    return <LoadingIcon state={LoadingState.Loading} />;
  }

  return (
    <View>
      <ImageZoom
        cropWidth={Dimensions.get('window').width}
        cropHeight={Dimensions.get('window').height}
        imageWidth={Dimensions.get('window').width}
        imageHeight={Dimensions.get('window').height}
      >
        <Image
          style={{ width: '100%', height: '100%' }}
          source={snapshotUrl}
          resizeMode="contain"
        />
      </ImageZoom>
    </View>
  );
}
