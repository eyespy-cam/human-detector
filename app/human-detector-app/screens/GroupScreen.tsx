import {useState} from "react";
import Camera from "../classes/Camera";
import {ScrollView, StyleSheet, Text, TouchableOpacity, View} from "react-native";
import CameraSettingsButton from "../components/CameraSettingsButton";
import Group from "../classes/Group";

export default function GroupMenu( { navigation }: Props ) {
    var groupOne:Group = new Group("AAAAAA's Group", "99");
    var groupTwo:Group = new Group("BBBBB's Group", "725");
    var groupThree:Group = new Group("CCCCC's Group", "400");

    const[listOfCameras, setListOfCameras] = useState([groupOne, groupTwo, groupThree])

    const pressHandler = () => {
        navigation.navigate('Cameras')
        console.log(typeof navigation);
    }

    return (
        <View style={styles.container}>
            <ScrollView>
                { listOfCameras.map((item) => {
                    return (
                        <View key={item.groupId}>
                            <TouchableOpacity
                                style={styles.menuItem}
                                onPress ={pressHandler}
                            >
                                <Text style={styles.menuButtonText}> {item.groupName} </Text>
                                <CameraSettingsButton cameraId={item.groupId}></CameraSettingsButton>
                            </TouchableOpacity>
                        </View>
                    )
                })}

                <View key={'add-button'}>
                    <TouchableOpacity
                        style={[styles.menuItem, styles.addButtonItem]}
                        onPress = {() => {
                            const cameraList: Camera[] = [...listOfCameras];
                            //adding camera here test
                            cameraList.push(new Camera("test", "test", "test"));
                            setListOfCameras(cameraList);
                        }}
                    >
                        <Text style={styles.addButtonText}> + </Text>
                    </TouchableOpacity>
                </View>
            </ScrollView>
        </View>

        /*
         * Instead of doing FlatList, I might want to change this to a scrollable view
         */
        //   <SafeAreaView style={styles.container}>
        //     <FlatList
        //         data={listOfCameras}
        //         renderItem={renderItem}
        //     />
        //   </SafeAreaView>
        //
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        //paddingTop: 40,
        //paddingHorizontal: 20
    },
    textInput: {
        borderWidth: 1,
        borderColor: '#777',
        padding: 8,
    },
    pads: {
        padding: 10
    },
    boldHeader: {
        fontWeight: 'bold',
        fontSize: 20,
    },
    menuItem: {
        marginTop: 24,
        marginLeft: 20,
        marginRight: 20,
        padding: 30,
        backgroundColor: '#E0FFFF',
        fontSize: 24,
        borderWidth: 2,
        borderColor: '#D3D3D3'
    },
    addButtonItem: {
        borderColor: '#D3D3D3',
        backgroundColor: '#DCDCDC',
    },
    menuButtonText: {
        fontSize: 24,
        marginTop: 10,
        marginBottom: 10,
        justifyContent: 'center',
        alignItems: 'center',
    },
    addButtonText: {
        alignItems: 'center',
        fontSize: 50,
        marginTop: 0,
        marginBottom: 0,
    },
});

export function GroupOnPress(){
    return (
        <View>
            <Text> Placeholder </Text>
        </View>
    )
}

export function GroupDisplayButton(){
    return (
        <View>
            <Text> Placeholder </Text>
        </View>
    )
}