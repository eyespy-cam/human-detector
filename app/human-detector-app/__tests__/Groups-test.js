import * as React from 'react';
import renderer from 'react-test-renderer';
import { Alert, Button } from 'react-native';
import { GroupOnPress, GroupDisplayButton } from '../screens/GroupScreen';
import Group from '../screens/GroupScreen';


//Method tests
it('GroupScreen Test 1: GroupOnPress() no cameras', () => {
    const component = renderer.create(GroupOnPress());
    expect(component.toJSON()).toMatchSnapshot();
})


it('GroupButton Test 2: GroupOnPress() 1 Camera', () => {
    const component = renderer.create(GroupOnPress());
    expect(component.toJSON()).toMatchSnapshot();
})

it('GroupButton Test 3: GroupOnPress() 5 Cameras', () => {
    const component = renderer.create(GroupOnPress());
    expect(component.toJSON()).toMatchSnapshot();
})

it('GroupButton Test 4: GroupDisplayButton() empty group', () => {
    const component = renderer.create(GroupOnPress());
    expect(component.toJSON()).toMatchSnapshot();
})

it('GroupScreen Test 5: GroupDisplayButton regular group', () => {
    const component = renderer.create(GroupOnPress());
    expect(component.toJSON()).toMatchSnapshot();
})