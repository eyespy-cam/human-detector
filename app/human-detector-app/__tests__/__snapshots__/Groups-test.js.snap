// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupButton Test 2: GroupOnPress() 1 Camera 1`] = `
<View>
  <Text>
     Placeholder text
  </Text>
</View>
`;

exports[`GroupButton Test 3: GroupOnPress() 5 Cameras 1`] = `
<View>
  <Text>
     Placeholder text
  </Text>
</View>
`;

exports[`GroupButton Test 4: GroupDisplayButton() empty group 1`] = `
<View>
  <Text>
     Placeholder text
  </Text>
</View>
`;

exports[`GroupScreen Test 1: GroupOnPress() no cameras 1`] = `
<View>
  <Text>
     Placeholder text
  </Text>
</View>
`;

exports[`GroupScreen Test 5: GroupDisplayButton regular group 1`] = `
<View>
  <Text>
     Placeholder text
  </Text>
</View>
`;

exports[`display group correctly 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Test test
    </Text>
  </View>
</View>
`;
